lib/services/utils/A6sRailwayUtil.ts

```diff
@@ -134,6 +143,29 @@ export class A6sRailwayUtil {
         return options;
     }
 
+    async resolveTree(station: IRailWayStation, pwd: string, graphPath = '') {
+        if (station.name !== 'a6s.external') {
+            if (Array.isArray(station.options)) {
+                station.options = await Promise.all(
+                    station.options.map(async (option: any) => {
+                        return await this.resolveTree(option, pwd, `${graphPath}->${station.name}->${option.name}`);
+                    })
+                );
+            }
+        } else {
+            const file = path.join(pwd, station.options.file);
+            const fileContent = await this.readYamlFile(file);
+
+            station.options = {
+                station: await this.resolveTree(fileContent.station, pwd, `${graphPath}->${station.name}`)
```

Comment:
Why do you need to concatenate strings? Represent the same with objects / arrays.